eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -38,26 +42,33 @@ steps:
 
   - template: /eng/common/TestResources/setup-environments.yml
 
-  - pwsh: |
-      eng/common/scripts/Import-AzModules.ps1
-
-      $subscriptionConfiguration = @'
-        ${{ parameters.SubscriptionConfiguration }}
-      '@ | ConvertFrom-Json -AsHashtable;
-
-      # The subscriptionConfiguration may have ArmTemplateParameters defined, so
-      # pass those in via the ArmTemplateParameters flag, and handle any
-      # additional parameters from the pipelines via AdditionalParameters
-      eng/common/TestResources/New-TestResources.ps1 `
-        -ResourceType '${{ parameters.ResourceType }}' `
-        -ServiceDirectory '${{ parameters.ServiceDirectory }}' `
-        -Location '${{ parameters.Location }}' `
-        -DeleteAfterHours '${{ parameters.DeleteAfterHours }}' `
-        @subscriptionConfiguration `
-        -AdditionalParameters ${{ parameters.ArmTemplateParameters }} `
-        -CI `
-        -Force `
-        -Verbose | Out-Null
+  - task: AzurePowerShell@5
     displayName: Deploy test resources
     env:
       TEMP: $(Agent.TempDirectory)
+    inputs:
+      azureSubscription: ${{ parameters.ServiceConnection }}
```

Comment:
TODO: only use `azureSubscription` if `ServiceConnection` is specified.

Reply:
If that is empty won't the AzurePowerShell step fail?